python/MufluxDigiReco.py

```diff
@@ -151,6 +189,115 @@ def digitize(self):
+	    muon_box = self.sGeo.GetTopVolume().FindNode("VMuonBox_1")
+	    muonbox_matrix = muon_box.GetMatrix()
+	    
+	    #other varialbles 
+            MuonTaggerTrackID = MuonTaggerHit.GetTrackID()
+            pid = MuonTaggerHit.PdgCode()
+
+	    #translation from top to MuonBox_1
+	    point = array('d', [MuonTaggerHit.GetX(), MuonTaggerHit.GetY(), MuonTaggerHit.GetZ(), 1])
+            point_muonbox = array('d', [0,0,0, 1])
+            muonbox_matrix.MasterToLocal(point, point_muonbox)
+
+	    #translation to local frame
+	    #point_muonbox = array('d', [MuonTaggerHit.GetX(), MuonTaggerHit.GetY(), MuonTaggerHit.GetZ()])
+	    point_local = array('d', [0,0,0, 1])
+            master_matrix.MasterToLocal(point_muonbox, point_local)
+            
+	    xcoord = point_local[0]
+	    ycoord = point_local[1]
+	   	
+            #selecting muons
+            if abs(pid) == 13 :
+                #identify individual rpcs
+		station = int(rpc[-1])
+		#if station > 2:
+			#station -= 1 #offset - 6 volumes defined in the geo file but 1 not sensitive
+		assert station in range(1,6) #limiting the range of rpcs
+		
+		#z coordinates of every station 
+		if station == 1: zcoord = 874.25
+		elif station == 2: zcoord = 959.25
+		elif station == 3: zcoord = 1004.25
+		elif station == 4: zcoord = 1049.25
+		elif station == 5: zcoord = 1094.25
+
+                #calculate strip
+                # x gives vertical direction
+                direction = 1
+                strip = StripX(xcoord) 
+		if strip < 0 and strip > 185 :
+			break 
+		#sampling number of strips around the exact strip for emulating clustering 
+		s = np.random.poisson(3)
+		strip = strip - int(s/2)
+		for i in range(0, s):
+			detectorid = station*10000 + direction*1000 + strip + i 
+			DetectorID.add(detectorid)
+		
+                #y gives horizontal direction
+                direction = 0
+                strip = StripY(ycoord)
+		if strip < 0 and strip > 117:
+			break
+		#sampling number of strips around the exact strip for emulating clustering 
+		s = np.random.poisson(3) 
+		strip = strip - int(s/2)
+		for i in range(0, s):
+			detectorid = station*10000 + direction*1000 + strip + i
+			DetectorID.add(detectorid)
+					
+
+	j = 0
+
+	for detID in DetectorID:
+		Hit = ROOT.MuonTaggerHit(detID, 0)
+		if self.digiMuonTagger.GetSize() == j: self.digiMuonTagger.Expand(j +1000)
+		self.digiMuonTagger[j] = Hit
+	
+	#cluster size loop - plotting the cluster size distribution 
+	cluster_size = list()
+	DetectorID_list = list(DetectorID) # turn set into list to allow indexing 
+	DetectorID_list.sort() # sorting the list
+	print DetectorID_list
+	if len(DetectorID_list) > 1:
+		
+		clusters = [[DetectorID_list[0]]]
+		for x in DetectorID_list[1:]:
+			if abs(x - clusters[-1][-1]) <= 1: 
+				clusters[-1].append(x)
+			else: 
+				clusters.append([x])
+		cluster_size = [len(x) for x in clusters]
+		print cluster_size
+		for i in cluster_size:
+			rc = h['muontagger_clusters'].Fill(i)
```

Comment:
Could you add some example histograms to the pull request for the clusters? I find this counting algorithm hard to follow.